db/compaction_job.cc

```diff
@@ -278,8 +278,9 @@ CompactionJob::CompactionJob(
       compaction_stats_(1),
       dbname_(dbname),
       db_options_(db_options),
-      env_options_(env_options),
       env_(db_options.env),
+      env_options_(
+          env_->OptimizeForCompactionTableRead(env_options, db_options_)),
```

Comment:
`OptimizeForCompactionTableRead()` have been called in `DBImpl`. No need to call here. Add a comment to explain instead?

Reply:
`CompactionJob` calls `TableCache::NewIterator`. One in `FinishCompactionOutputFile` and another in `ProcessKeyValueCompaction`. The former needs an original `env_options` and the latter needs an "optimized" version. With the introduction of `CompactionJob:: env_options_for_compaction_`, `DBImpl::env_options_for_compaction_` is probably redundant but I prefer to remove that in a separate PR since 1) to keep this PR small, 2) FlushJob also needs to be updated.